arangod/Cluster/MaintenanceFeature.cpp

```diff
@@ -947,3 +950,42 @@ void MaintenanceFeature::proceed() {
   _pauseUntil = std::chrono::steady_clock::duration::zero();
 }
 
+std::shared_ptr<maintenance::ActionDescription> MaintenanceFeature::isShardLocked(ShardID const& shardId) const {
+  auto it = _shardActionMap.find(shardId);
+  if (it == _shardActionMap.end()) {
+    return nullptr;
+  }
+  return it->second;
+}
+
+bool MaintenanceFeature::lockShard(ShardID const& shardId,
+                                   std::shared_ptr<maintenance::ActionDescription> const& description) {
+  LOG_TOPIC("aaed2", DEBUG, Logger::MAINTENANCE) << "Locking shard " << shardId << " for action " << *description;
+  std::lock_guard<std::mutex> guard(_shardActionMapMutex);
+  auto pair = _shardActionMap.emplace(shardId, description);
+  return pair.second;
+}
+
+bool MaintenanceFeature::unlockShard(ShardID const& shardId) {
```

Comment:
This is still returning a boolean value which is never checked at the call sites.
So we can change the function signature to return void, and call `erase` unconditionally.

Reply:
Note that we must not pass the `end()` iterator to `erase`, so I think a check remains sane, even when we don't need the return value. If we expect a lock to exist when unlocking, a log message in that branch might be helpful, too.